.github/workflows/sc4s-stable.yml

```diff
@@ -0,0 +1,179 @@
+name: SC4S stable
+
+on:
+  workflow_dispatch:
+    inputs:
+      owner:
+        description: 'SC4S repository owner'
+        required: true
+        default: 'splunk'
+        type: string
+      repo:
+        description: 'SC4S repository'
+        required: true
+        default: 'splunk-connect-for-syslog'
+        type: string
+      ref:
+        description: 'SC4S ref (branch, tag, commit id)'
+        required: false
+        type: string
+      syslog-ng-image-version:
+        description: 'syslog-ng image version'
+        required: true
+        default: 'nightly' # until https://github.com/syslog-ng/syslog-ng/pull/4390 is released
+        #default: 'latest'
+        type: string
+  pull_request:
+    paths:
+      - 'sc4s/**'
+      - '.github/workflows/**'
+  push:
+    paths:
+      - 'sc4s/**'
+      - '.github/workflows/**'
+    branches:
+      - '**'
+    tags:
+      - 'sc4s-*'
+
+env:
+  DOCKER_IMAGE_NAME: ghcr.io/axoflow/syslog-ng-splunk-connector
+  PLATFORMS: linux/amd64,linux/arm64
+
+jobs:
+  detect-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.ref }}
+    steps:
+      - name: Detect SC4S version
+        id: version
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const input_ref = '${{ inputs.ref }}';
+            if (input_ref) {
+              console.log('Version from workflow input: ' + input_ref);
+              core.setOutput('ref', input_ref);
+              return;
+            }
+
+            if (context.ref.startsWith('refs/tags/sc4s-')) {
+              const tag = "v" + context.ref.replace("refs/tags/sc4s-", "");
+              console.log('Version from tag: ' + tag);
+              core.setOutput('ref', tag);
+              return;
+            }
+
+            const owner = '${{ inputs.owner || 'splunk' }}';
```

Comment:
[optional]

Can we factor out the fallback default values to `env:`, and only reference them here? Or maybe we could work solely from environment variables every time, and we could have a prepare step, which fills them from the input, if we are running on `workflow_distpatch`, else the hard coded defaults.

It is easy to forget to update the default values, if they are scattered in the workflow file.

Reply:
Good idea.

Reply:
Fixed.